kratos/utilities/math_utils.h

```diff
@@ -403,7 +403,31 @@ class MathUtils
             noalias(InvertedMatrix) = prod(auxInv, trans(InputMatrix));
         }
     }
-    
+
+    /**
+     * @brief This function is designed to be called when a dense linear system is needed to be solved
+     * @param rA System matrix
+     * @param rX Solution vector. it's also the initial guess for iterative linear solvers.
+     * @param rB Right hand side vector.
+     */
+    static void Solve(
+        const MatrixType& rA,
```

Comment:
it is more elegant if instead of making the copy internally you pass A by value

Reply:
OK

Reply:
The lu_substitute modifies internally A, so better to do a copy, imagine that you are interested in keep the A matrix as it is